manager/controlapi/secret.go

```diff
@@ -31,6 +31,34 @@ func secretFromSecretSpec(spec *api.SecretSpec) *api.Secret {
 	}
 }
 
+// checkSecretInUse does a best effort to find if the passed in secret is in
+// use by any services
+func (s *Server) checkSecretInUse(secretID string) error {
+	var (
+		services []*api.Service
+		err      error
+	)
+
+	s.store.View(func(tx store.ReadTx) {
```

Comment:
This is how all of our checks work, and why there is a description that this is "best-effort"